pkg/operator/resource/resourcemerge/generic_config_merger.go

```diff
@@ -228,3 +236,36 @@ func intersectValue(x1, x2 interface{}) interface{} {
 		return x1
 	}
 }
+
+// IsRequiredConfigPresent can check an observedConfig to see if certain required paths are present in that config.
+// This allows operators to require certain configuration to be observed before proceeding to honor a configuration or roll it out.
+func IsRequiredConfigPresent(config []byte, requiredPaths [][]string) error {
+	if len(config) == 0 {
+		return fmt.Errorf("no observedConfig")
+	}
+
+	existingConfig := map[string]interface{}{}
+	if err := json.NewDecoder(bytes.NewBuffer(config)).Decode(&existingConfig); err != nil {
+		return fmt.Errorf("error parsing config, %v", err)
+	}
+
+	for _, requiredPath := range requiredPaths {
+		configVal, found, err := unstructured.NestedFieldNoCopy(existingConfig, requiredPath...)
+		if err != nil {
+			return fmt.Errorf("error reading %v from config, %v", strings.Join(requiredPath, "."), err)
+		}
+		if !found {
+			return fmt.Errorf("%v missing from config", strings.Join(requiredPath, "."))
+		}
+		if configVal == nil {
+			return fmt.Errorf("%v null in config", strings.Join(requiredPath, "."))
+		}
+		if configValSlice, ok := configVal.([]interface{}); ok && len(configValSlice) == 0 {
```

Comment:
the last two cases don't match "Present" IMO. Then better call it `IsRequiredConfigNonEmpty`.

Reply:
> the last two cases don't match "Present" IMO. Then better call it `IsRequiredConfigNonEmpty`.

That's what the word present means.